src/features/locators/components/LocatorEditDialog.tsx

```diff
@@ -215,6 +228,35 @@ export const LocatorEditDialog: React.FC<Props> = ({
     updateRef(OnbrdStep.EditLocator, undefined, isOkButtonDisabled ? undefined : handleCreateCustomLocator);
   };
 
+  const navigateToOriginalLocator = async (event: MouseEvent<HTMLSpanElement>) => {
+    event.stopPropagation();
+    const _locator = form.getFieldValue("locator");
+    form.resetFields();
+    setIsModalOpen(false);
+
+    let locatorValue;
+    locatorType === LocatorType.cssSelector
+      ? (locatorValue = await evaluateCssSelector(_locator, element_id, jdnHash))
+      : (locatorValue = await evaluateXpath(_locator, element_id, jdnHash));
+
+    const { element_id: _element_id, foundHash } = JSON.parse(locatorValue);
+    const duplicates = checkDuplicates(foundHash, locators, _element_id);
+    if (duplicates.length) {
+      dispatch(setActiveSingle(duplicates[0]));
+      dispatch(setScrollToLocator(duplicates[0].element_id));
+    }
+  };
```

Comment:
please, extract this logic to a new function and place it outside of a component. it can be a thunk